recipes/repository.rb

```diff
@@ -19,9 +19,29 @@
 #
 
 # Official Zabbix repository
-apt_repository 'zabbix' do
-  uri 'http://repo.zabbix.com/zabbix/2.4/ubuntu'
-  distribution node['lsb']['codename']
-  components %w(main)
-  key 'http://repo.zabbix.com/zabbix-official-repo.key'
+case node['platform']
+when 'debian'
+  apt_repository 'zabbix' do
+    uri node['zabbix_ng']['repository']['uri']
+    distribution node['lsb']['codename']
+    components %w(main)
+    key node['zabbix_ng']['repository']['key']
+  end
+when 'ubuntu'
+  apt_repository 'zabbix' do
+    uri node['zabbix_ng']['repository']['uri']
+    distribution node['lsb']['codename']
+    components %w(main)
+    key node['zabbix_ng']['repository']['key']
+  end
+when 'redhat', 'centos', 'fedora'
+  yum_repository 'zabbix' do
+    description "Official zabbix repository"
+    baseurl node['zabbix_ng']['repository']['uri']
+    gpgkey node['zabbix_ng']['repository']['key']
+  end
+  include_recipe 'yum-epel'
```

Comment:
Maybe a default case should be inserted, which prints some sort of error message if this cookbook is run on a platform that is not supported? 
It's not that important though, as for other platforms the package names are probably different as well and the cookbook will fail anyway.